sound/soc/sof/probe.c

```diff
@@ -0,0 +1,205 @@
+	return ret;
+}
+EXPORT_SYMBOL(sof_ipc_probe_get_dma);
+
+int sof_ipc_probe_dma_attach(struct snd_sof_dev *sdev,
+		struct sof_probe_dma *dma, size_t num_dma)
+{
+	struct sof_ipc_probe_dma_set_params *msg;
+	struct sof_ipc_reply reply;
+	size_t bytes = sizeof(*dma) * num_dma;
+	int ret;
+
+	msg = kzalloc(sizeof(*msg) + bytes, GFP_KERNEL);
+	if (!msg)
+		return -ENOMEM;
+	msg->hdr.size = sizeof(*msg) + bytes;
+	msg->hdr.cmd = SOF_IPC_GLB_PROBE | SOF_IPC_PROBE_DMA_SET;
+	memcpy(&msg->dma[0], dma, bytes);
+
+	ret = sof_ipc_tx_message(sdev->ipc, msg->hdr.cmd, msg, msg->hdr.size,
+			&reply, sizeof(reply));
+	kfree(msg);
+	return ret;
+}
+EXPORT_SYMBOL(sof_ipc_probe_dma_attach);
+
+int sof_ipc_probe_dma_detach(struct snd_sof_dev *sdev,
+		unsigned int *stream_tag, size_t num_stream_tag)
+{
+	struct sof_ipc_probe_dma_detach_params *msg;
+	struct sof_ipc_reply reply;
+	size_t bytes = sizeof(*stream_tag) * num_stream_tag;
+	int ret;
+
+	msg = kzalloc(sizeof(*msg) + bytes, GFP_KERNEL);
+	if (!msg)
+		return -ENOMEM;
+	msg->hdr.size = sizeof(*msg) + bytes;
+	msg->hdr.cmd = SOF_IPC_GLB_PROBE | SOF_IPC_PROBE_DMA_DETACH;
+	memcpy(&msg->stream_tag[0], stream_tag, bytes);
+
+	ret = sof_ipc_tx_message(sdev->ipc, msg->hdr.cmd, msg, msg->hdr.size,
+			&reply, sizeof(reply));
+	kfree(msg);
+	return ret;
+}
+EXPORT_SYMBOL(sof_ipc_probe_dma_detach);
+
+int sof_ipc_probe_get_points(struct snd_sof_dev *sdev,
+		struct sof_probe_point_desc **desc, size_t *num_desc)
+{
+	int ret;
+
+	ret = sof_ipc_probe_get(sdev, SOF_IPC_PROBE_POINT_GET,
+			(void **)desc, num_desc);
+	*num_desc /= sizeof(**desc);
+	return ret;
+}
+EXPORT_SYMBOL(sof_ipc_probe_get_points);
+
+int sof_ipc_probe_points_connect(struct snd_sof_dev *sdev,
+		struct sof_probe_point_desc *desc, size_t num_desc)
+{
+	struct sof_ipc_probe_point_set_params *msg;
+	struct sof_ipc_reply reply;
+	size_t bytes = sizeof(*desc) * num_desc;
+	int ret;
+
+	msg = kzalloc(sizeof(*msg) + bytes, GFP_KERNEL);
+	if (!msg)
+		return -ENOMEM;
+	msg->hdr.size = sizeof(*msg) + bytes;
+	msg->hdr.cmd = SOF_IPC_GLB_PROBE | SOF_IPC_PROBE_POINT_SET;
+	memcpy(&msg->desc[0], desc, bytes);
+
+	ret = sof_ipc_tx_message(sdev->ipc, msg->hdr.cmd, msg, msg->hdr.size,
+			&reply, sizeof(reply));
+	kfree(msg);
+	return ret;
+}
+EXPORT_SYMBOL(sof_ipc_probe_points_connect);
+
+int sof_ipc_probe_points_disconnect(struct snd_sof_dev *sdev,
+		unsigned int *buffer_id, size_t num_buffer_id)
+{
+	struct sof_ipc_probe_point_remove_params *msg;
+	struct sof_ipc_reply reply;
+	size_t bytes = sizeof(*buffer_id) * num_buffer_id;
+	int ret;
+
+	msg = kzalloc(sizeof(*msg) + bytes, GFP_KERNEL);
+	if (!msg)
+		return -ENOMEM;
+	msg->hdr.size = sizeof(*msg) + bytes;
+	msg->hdr.cmd = SOF_IPC_GLB_PROBE | SOF_IPC_PROBE_POINT_REMOVE;
+	memcpy(&msg->buffer_id[0], buffer_id, bytes);
+
+	ret = sof_ipc_tx_message(sdev->ipc, msg->hdr.cmd, msg, msg->hdr.size,
+			&reply, sizeof(reply));
+	kfree(msg);
+	return ret;
+}
+EXPORT_SYMBOL(sof_ipc_probe_points_disconnect);
```

Comment:
@crojewsk overall this file looks fine but there are no comments at all. Could you please add some comments for each method maybe?

Reply:
Heh : D
When it comes to programming, one could say code style is like religion.
Let's say I'm representing more radical group. Readable code does not equal code throttled with comments and newlines.

I agree with adding description ABOVE each API method, but I'm against adding multiple comments within code itself. Locals/ callbacks names should be self-explanatory.

Reply:
Fixed in v5.